cdap-common/src/main/java/io/cdap/cdap/common/lang/InstantiatorFactory.java

```diff
@@ -174,14 +181,28 @@ public T create() {
   }
 
   private <T> Instantiator<T> getByUnsafe(final TypeToken<T> type) {
+    //We do introspection once and then just use reflection to set field values
+    List<Map.Entry<Field, Object>> fieldsToInitialize = new ArrayList<>();
+    Reflections.visit(null, type.getType(), new FieldVisitor() {
+      @Override
+      public void visit(Object instance, Type inspectType, Type declareType, Field field) throws Exception {
+        if (Modifier.isStatic(field.getModifiers())) {
+          return;
+        }
+        fieldsToInitialize.add(new AbstractMap.SimpleEntry<>(field, Defaults.defaultValue(field.getType())));
+      }
+    });
+
     return new Instantiator<T>() {
       @Override
       public T create() {
         try {
           Object instance = UNSAFE.allocateInstance(type.getRawType());
-          Reflections.visit(instance, type.getType(), new FieldInitializer());
+          for (Map.Entry<Field, Object> e : fieldsToInitialize) {
+            e.getKey().set(instance, e.getValue());
+          }
           return (T) instance;
-        } catch (InstantiationException e) {
+        } catch (InstantiationException | IllegalAccessException e) {
           throw Throwables.propagate(e);
```

Comment:
is this single pipe intentional?

Reply:
That's multicatch syntax, see https://www.geeksforgeeks.org/multicatch-in-java/

Reply:
Yes, this allows us to manipulate multiple exception types using the same code https://docs.oracle.com/javase/8/docs/technotes/guides/language/catch-multiple.html#multiple

Reply:
ah I see, thanks!